app/controllers/api/v1/merchants_controller.rb

```diff
@@ -4,10 +4,34 @@ def index
         serialized_merchants = MerchantSerializer.new(merchants)
         render json: serialized_merchants
     end 
-
+    
     def show
-        items = Item.where(merchant_id: params[:id])
-        serialized_items = ItemSerializer.new(items)
-        render json: serialized_items
+        merchant = Merchant.find(params[:merch_id])
+        serialized_merchant = MerchantSerializer.new(merchant)
+        render json: serialized_merchant
+    end 
+
+    def find
+        merchant = Merchant.find_by(merch_params)
+        serialized_merchant = MerchantSerializer.new(merchant)
+        render json: serialized_merchant
+    end 
+
+    def find_all
+        merchants = Merchant.where(merch_params)
+        serialized_merchants = MerchantSerializer.new(merchants)
+        render json: serialized_merchants
+    end 
+
+    def random
+        merchant = Merchant.all.sample
+        serialized_merchant = MerchantSerializer.new(merchant)
+        render json: serialized_merchant
```

Comment:
You are repeating these two lines in every action. Consider breaking these out to a method?

```ruby
def render_merchants(merchants)
    serialized_merchants = MerchantSerializer.new(merchants)
    render json: serialized_merchants
end
```